plugins/outline_creator.js

```diff
@@ -101,7 +97,28 @@ function createOutline(outline_thickness) {
                 }
             }
         }).init();
-    };
+    }
+
+    // Mesh handling
+    for (const mesh of Mesh.selected) {
+        mesh.duplicate();
+        mesh.oldVertices = {};
+
+        for (const key in mesh.vertices) {
+            mesh.oldVertices[key] = mesh.vertices[key].slice();
+        }
+
+        mesh.forAllFaces(face => {
+            face.invert();
+        })
+
+        mesh.resize(outline_thickness * 2, 0, false, false, true);
+        mesh.resize(outline_thickness, 1, false, false, true);
+        mesh.resize(outline_thickness * 2, 2, false, false, true);
+        mesh.name = mesh.name + "_outline";
+    }
+
+    Canvas.updateAll();

```

Comment:
Canvas.updateAll is quite inefficient and will be deprecated at some point. I'd recommend to use Canvas.updateView API instead, with all affected elements and the transform and geometry aspects: https://github.com/JannisX11/blockbench/blob/master/js/modeling/transform.js#L676